lib/spack/spack/test/package_sanity.py

```diff
@@ -173,13 +173,20 @@ def test_prs_update_old_api():
     failing = []
     for file in changed_package_files:
         name = os.path.basename(os.path.dirname(file))
-        pkg = spack.repo.get(name)
 
-        # Check for old APIs
-        failed = (hasattr(pkg, 'setup_environment') or
-                  hasattr(pkg, 'setup_dependent_environment'))
-        if failed:
-            failing.append(pkg)
+        try:
+            pkg = spack.repo.get(name)
+        except spack.repo.UnknownPackageError:
+            # This package is from the mock repo; does not necessarily require
```

Comment:
This seems like an inference that could go wrong. Is it possible to add a fixture to try the builtin.mock repo, or to check that specifically when there is a failure? Are there `builtin.mock` packages that should be using the old API?

Reply:
I don't want to assume all `builtin.mock` packages are using the new API, as they could be testing the backwards compatibility to the old API.

Reply:
There's a fixture to use the mock repo, but then we wouldn't be able to test things in the main repo. And if we made a fixture to have both active, we wouldn't be able to test things in the lower priority repo that shadow things in the higher priority repo.

I think it's safer to make this assumption.

Reply:
you can alternate between both repos, or perhaps there's a way to check the mock repo for the existence of the package without adding it to the global repo dir?

Reply:
We have the filenames already, what about literally just scanning the text for those strings?

Reply:
That would definitely be better (and I think that's sufficient).

Reply:
Done